### PR TITLE
Java attacher runs only in non-cloud environments

### PR DESCRIPTION
**Pull request by stuartnelson3**
## Motivation/summary

start the java attacher jar only when running in a non-cloud environment

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
- [ ] Documentation has been updated

## How to test these changes

1. start apm-server with the java attacher config enabled and  `ELASTIC_AGENT_CLOUD` set in the environment
2. verify the java attacher does not start
3. start apm-server with the java attacher config enabled and  `ELASTIC_AGENT_CLOUD` NOT set in the environment
4. verify the java attacher does start

## Related issues

requested in https://github.com/elastic/apm-server/issues/5479
depends on https://github.com/elastic/apm-server/pull/6593

### CODE REVIEW

File: beater/beater.go [hunk 1]
```diff
@@ -435,6 +437,15 @@ func (s *serverRunner) run(listener net.Listener) error {
...
 	}
 
+	if eac == "" {
+		// We aren't running in a cloud environment
+		attacher, err := javaattacher.New(s.config.JavaAttacherConfig)
+		if err != nil {
+			s.logger.Errorf("failed to start java attacher: %v", err)
+		}
+		go attacher.Run(s.runServerContext)
+	}
```
Comment (felixbarny) [inline beater/beater.go:447]: Maybe log an error if trying to start the attacher on cloud.

File: beater/beater.go [hunk 2]
```diff
@@ -426,7 +427,8 @@ func (s *serverRunner) run(listener net.Listener) error {
 	cfg := ucfg.Config(*s.rawConfig)
 	parentCfg := cfg.Parent()
 	// Check for an environment variable set when running in a cloud environment
-	if eac := os.Getenv("ELASTIC_AGENT_CLOUD"); eac != "" && s.config.Kibana.Enabled {
+	eac := os.Getenv("ELASTIC_AGENT_CLOUD")
```
Comment (felixbarny) [inline beater/beater.go:430]: Is `ELASTIC_AGENT_CLOUD` also set on ECE?

Comment (simitt) [reply beater/beater.go:430]: yes

**Comment by stuartnelson3** [comment]
@felixbarny would you be able to test this? the docker container doesn't contain the java runtime so the attacher errors out, but either building a tarball or running `make && make build/apm-agent-attach-cli-1.27.0-slim.jar && cp build/java-attacher.jar .` should put the .jar in the correct place relative to the apm-server binary.

**Comment by felixbarny** [comment]
We have a dedicated task to test the attachment end-to-end: https://github.com/elastic/apm-agent-java/issues/2196. I think it makes sense to start that after https://github.com/elastic/apm-server/issues/4830 is closed.

Sub-tasks like these can tested and reviewed in isolation as good as possible. I think the testing phase can start in parallel with the UI implementation. After the UI is done, there'll be a second QA phase that tests fully end-to-end. 

WDYT?

**Review by felixbarny — APPROVED** [review]

**Comment by stuartnelson3** [comment]
Sounds good to me, will merge once tests pass

**Comment by marclop** [comment]
Verified with cdeec66e1:

```json
{"log.level":"error","@timestamp":"2022-01-04T15:41:28.521+0800","log.logger":"beater","log.origin":{"file.name":"beater/beater.go","file.line":443},"message":"java attacher not supported in cloud environments","service.name":"apm-server","ecs.version":"1.6.0"}
```

**Pull request merged**